packages/fxa-settings/src/lib/test-utils.ts

```diff
@@ -25,24 +25,24 @@ export const typeByLabelText = (labelText: string) => async (x: string) => {
 type LinkParamValue = string | null;
 
 export const getSearchWithParams = ({
-  mockToken,
-  mockCode,
   mockEmail,
   mockPasswordHash,
+  mockToken,
+  mockCode,
   mockUid,
 }: {
-  mockToken: LinkParamValue;
-  mockCode: LinkParamValue;
   mockEmail: LinkParamValue;
   mockPasswordHash?: LinkParamValue;
+  mockToken: LinkParamValue;
+  mockCode: LinkParamValue;
   mockUid?: LinkParamValue;
 }) => {
-  const tokenParam = mockToken !== null ? `token=${mockToken}&` : '';
-  const codeParam = mockCode !== null ? `code=${mockCode}&` : '';
   const emailParam = mockEmail !== null ? `email=${mockEmail}&` : '';
   const passwordHashParam = mockPasswordHash
-    ? `emailToHashWith=${mockPasswordHash}`
+    ? `emailToHashWith=${mockPasswordHash}&`
     : '';
+  const tokenParam = mockToken !== null ? `token=${mockToken}&` : '';
+  const codeParam = mockCode !== null ? `code=${mockCode}&` : '';
   const uidParam = mockUid ? `uid=${mockUid}` : '';
-  return `?${tokenParam}${codeParam}${emailParam}${passwordHashParam}${uidParam}`;
+  return `?${emailParam}${passwordHashParam}${tokenParam}${codeParam}${uidParam}`;
```

Comment:
Were the changes in this file needed? Looks like the order of a few things were swapped around 🤔

Reply:
The tests were failing because the order of the params was different than the actual navigation call. Perhaps that calls for a more flexible test vs changes here 🤔

Reply:
This is well out of scope, but wonder if some utility method to convert an object to a query string (while stripping `undefined` values) would be cleaner:

```js
const mockEmail = "peter@foo.bar";
const mockPasswordHash = "beep.boop";
const mockToken = "FoObAr";
let mockCode;
let mockUid;

const query = queryParamsFromObj({
  email: mockEmail,
  emailToHashWith: mockPasswordHash,
  token: mockToken,
  uid: mockUid,
  code: mockCode,
});
console.log(query); // "email=peter%40foo.bar&emailToHashWith=beep.boop&token=FoObAr"

function queryParamsFromObj(obj = {}) {
  return Object.entries(obj)
      .reduce((acc, [k, v]) => {
        v !== undefined && acc.set(k, v);
        return acc;
      }, new URLSearchParams())
      .toString();
}
```

Reply:
@pdehaan probably so. We already have a couple related tickets filed to make our query param references consistent, that might be a good place to do this, or when we rework the reliers (which deal with these params) [into integrations](https://mozilla-hub.atlassian.net/browse/FXA-7308). Feels like we need to clean up a lot of testing around location, I'll add a note in that ticket to either do it there or add a note to some follow up to.

Reply:
Turns out we no longer needed this original util, so I've gone ahead and removed it. 🧹